policy/predicate/file.go

```diff
@@ -181,27 +197,146 @@ func (pred *NoChangedFiles) Trigger() common.Trigger {
 	return common.TriggerCommit
 }
 
-type FileNotDeleted struct {
+type FileAdded struct {
 	Paths []common.Regexp `yaml:"paths"`
 }
 
-var _ Predicate = &FileNotDeleted{}
+var _ Predicate = &FileAdded{}
 
-func (pred *FileNotDeleted) Evaluate(ctx context.Context, prctx pull.Context) (*common.PredicateResult, error) {
-	var paths []string
-	for _, r := range pred.Paths {
-		paths = append(paths, r.String())
+func (pred *FileAdded) Evaluate(ctx context.Context, prctx pull.Context) (*common.PredicateResult, error) {
+	paths := getPathStrings(pred.Paths)
+	predicateResult := initializePredicateResult(paths, "added", false, false)
+
+	changedFiles, err := prctx.ChangedFiles()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list changed files")
 	}
 
-	predicateResult := common.PredicateResult{
-		Satisfied:         true,
-		ValuePhrase:       "deleted files",
-		Values:            []string{},
-		ConditionPhrase:   "match path patterns",
-		ReverseSkipPhrase: true,
-		ConditionValues:   paths,
+	if len(changedFiles) == 0 {
+		predicateResult.Description = "No files were changed"
+		return &predicateResult, nil
+	}
```

Comment:
If we're looking to save lines, we can probably remove this check from every predicate. The rest of the function should behave the same and the "No added files match the specified patterns" message isn't wrong in this case. PRs that don't modify anything should also be rare / degenerate, so as long as nothing breaks, I don't think they require special handling. But it also doesn't hurt to leave it.